vllm/config.py

```diff
@@ -869,6 +869,13 @@ def __init__(
                                  f"distributed executor backend "
                                  f"'{self.distributed_executor_backend}'.")
 
+        if current_platform.is_tpu() and self.world_size > 1:
+            if self.distributed_executor_backend is None:
+                self.distributed_executor_backend = "ray"
```

Comment:
I mean, I think this line should be enough to change the default backend to ray in tpu case.

Reply:
Oh the error is for those who used `distributed_executor_backend="mp"`.

Reply:
why do we need to error if users explicitly specify the mp backend?

Reply:
The MP backend is not supported for TPUs at the moment. Without this line, the user will get the error:
```
"/vllm/engine/llm_engine.py", line 505, in _get_executor_cls
    assert distributed_executor_backend is None
```

Reply:
I'm confused by `Actually, the MP backend would also work for TPUs`

so MP backend for TPU is actually not implemented yet.

Reply:
Yes. We don't have a executor for tpu + mp.

Reply:
cc @njhill if you have any misunderstanding. it is because we currently only have ray backend supported in tpu.